static/src/javascripts/bootstraps/commercial.ts

```diff
@@ -0,0 +1,11 @@
+import { onConsent } from '@guardian/consent-management-platform';
+import { init as initConsented } from './commercial.consented';
+
+void onConsent().then((consentState) => {
```

Comment:
What's the delay here? Does this potentially impact time to render for an ad?

Reply:
Yes it will in its current form. The delay would be Sourcepoint initialising, CMP transforming raw consent state and invocation of the promise wrapper `onConsent`.

We currently already check consent before loading googletag (and other points) but that happens post-initialisation of `standalone.commercial.ts`. So it's not a new cost but in theory we are able to initialise a lot of more our stack before we get the consent callbacks that trigger further progress.

So we have the choice of:

- keeping our current stack and conditionally wrap consented/consentless modules
- create a new ad stack that will be conditionally loaded based on consent

We would like to explore the trade offs in creating a new bundle which in theory would be _much_ smaller and 'do less' than the current commercial bundle. We have talked about ways we could improve on needing consent up front e.g.

- bypass Sourcepoint and read the raw consent state ourselves
- storing a simpler boolean proxy for consent after the first invocation

That being said I wouldn't read too much into the current logic as this is a WIP spike to shake out any integration issues with a consentless advertising vendor.

Performance is not the main priority at the moment but once the implementation has settled we will be very looking at our metrics very closely for any regressions.